### ISSUE TITLE
catkin_test_results cannot parse test results with ansi escape codes

### ISSUE THREAD

**Issue by wjwwood** [ros/catkin#511]
When running `catkin_test_results test_results` I get:

```
Skipping "capabilities/rostest-build_test_rostest_demo.xml": not well-formed (invalid token): line 22, column 0
capabilities/MISSING-rostest-test_rostest_demo.xml: 1 tests, 0 errors, 1 failures
Summary: 25 tests, 0 errors, 1 failures
```

This is the file which gives the problem:

``` xml
<?xml version="1.0" encoding="utf-8"?>
<testsuite errors="0" failures="0" name="unittest.suite.TestSuite" tests="1" time="3.603">
  <testcase classname="rostest.runner.RosTest" name="testdemo" time="3.6032"></testcase>
  <system-out><![CDATA[[ROSTEST]setup[/Users/william/devel/capabilities/test/rostest/demo.test] run_id[05d93b91-2157-11e3-9bf7-542696cef915] starting
[1mstarted roslaunch server http://dosa:56314/[0m
ros_comm version 1.9.48


SUMMARY
========

PARAMETERS
 * /rosdistro
 * /rosversion

NODES
  /
    capability_server (capabilities/capability_server)

[ROSTEST]setup[/Users/william/devel/capabilities/test/rostest/demo.test] run_id[05d93b91-2157-11e3-9bf7-542696cef915] done
[ROSTEST]Running test [demo]
auto-starting new master
[1mprocess[master]: started with pid [10971][0m
[1mROS_MASTER_URI=http://dosa:22422/[0m
[1msetting /run_id to 05d93b91-2157-11e3-9bf7-542696cef915[0m
[1mprocess[rosout-1]: started with pid [10974][0m
started core service [/rosout]
[1mprocess[capability_server-2]: started with pid [10975][0m
[ROSTEST]running test demo
[1mprocess[demo-3]: started with pid [10976][0m
[1m[demo-3] process has finished cleanly
log file: /Users/william/.ros/log/05d93b91-2157-11e3-9bf7-542696cef915/demo-3*.log[0m
[ROSTEST]test [demo] finished
[ROSTEST]test [demo] results are in [/Users/william/devel/capabilities/build/test_results/capabilities/rosunit-demo.xml]
[ROSTEST]test [demo] results summary: 0 errors, 0 failures, 1 tests
[ROSTEST][ROSTEST] test [demo] done
[ROSTEST]tearDown[/Users/william/devel/capabilities/test/rostest/demo.test]
shutting down processing monitor...
[capability_server-2] killing on exit
[rosout-1] killing on exit
[master] killing on exit
... shutting down processing monitor complete
[ROSTEST]rostest teardown /Users/william/devel/capabilities/test/rostest/demo.test complete
]]></system-out>
  <system-err><![CDATA[]]></system-err>
</testsuite>
```

It looks like to me that the ANSI escape characters are causing the issue.

**Comment by dirk-thomas** [comment]
`catkin_test_results` will only handle valid XML input. It should probably be fixed during generation of the xml output (which I thought should actually alread happen but seems to not catch this input).

**Comment by wjwwood** [comment]
I am running on my Mac, perhaps the sanitization is not being done correctly on OS X, though I couldn't imagine how it would be different. I'll look into it.

**Comment by wjwwood** [comment]
When building out of source the file gets sanitized and this is no longer an issue. This probably means it only sanitizes files it expects to be there, but parses all xml files in the test_results folder?

See: #512

**Comment by dirk-thomas** [comment]
The actual fix will be in rosunit: ros/ros#35. Closing this ticket.
